### ISSUE TITLE
macOS build doesn't include MoltenVK library

### ISSUE THREAD

**Issue by designrad** [bjornbytes/lovr#610]
404 error when accessing https://lovr.org/f/L%D6VR.dmg.
Link from https://lovr.org/downloads -- MacOS

**Comment by bjornbytes** [comment]
Should be fixed, I just renamed it to lovr.dmg to sidestep the UTF-8 thing.  Thanks!

**Comment by designrad** [comment]
Downloaded dmg file "is damaged and can't be opened". Macbook Pro with Apple M1 chip, macOS Ventura 13.0.1. I retried a few times but it won't open the dmg. Not from disk, not from Applications folder.

**Comment by bjornbytes** [comment]
Hm, that's annoying.  You could try the nightly build at the bottom of this page: https://github.com/bjornbytes/lovr/actions/runs/3616216122

**Comment by designrad** [comment]
Doesn't work. Zip file gets uncompressed automatically into just a folder called "Contents". This is usually how macOS apps are structured. Changing to .app or just removing .zip doesn't work.

<img width="727" alt="Screenshot 2022-12-06 at 09 57 51" src="https://user-images.githubusercontent.com/23334059/205866100-4c42380e-6676-4f85-8cbc-2e72eaf9fb23.png">

**Comment by TrueGeek** [comment]
I'm having the same issue as @designrad . (Although my message says "You should move it to the trash" instead of "bin". 😀)

I downloaded the nightly build and made lovr executable with `chmod 701 lovr`. I get these error when running though:

```
Error:
GPU error: Failed to load vulkan library

Stack:
boot.lua:90: in function <boot.lua:4>
[C]: in function 'xpcall'
```

**Comment by wolf81** [comment]
@TrueGeek You could try to manually install the Vulkan SDK, this resolved the issue for me: https://github.com/bjornbytes/lovr/issues/615#issuecomment-1356658463

**Comment by TrueGeek** [comment]
@bjornbytes That worked, thanks! (sorry for the delay. Work has been keeping me busy)

**Comment by bjornbytes** [comment]
I now understand what needs to be done here:

- CMake: on macOS, needs to find and link against the MoltenVK runtime library (and some other frameworks).
- GitHub actions: should install the Vulkan SDK, possibly also build both x86_64/arm64 and create a universal build.

This way:

- People don't need the Vulkan SDK installed to use LÖVR.
- LÖVR will work on both x86_64 and arm64 macOS systems.

I found a random GitHub repo that has a good example of installing the Vulkan SDK on macOS GitHub actions:

https://github.com/nihui/vkpeak/blob/master/.github/workflows/release.yml#L68

**Comment by bjornbytes** [comment]
Made the following changes:

- Fixed the folder structure for macOS builds on GitHub actions.  So it should now extract properly as an .app.
- Made macOS builds link against Vulkan, and GitHub actions now installs the Vulkan SDK and links against MoltenVK, including it in the .app.  So the Vulkan SDK shouldn't be necessary anymore.
- Made the Vulkan backend first try linking against `libvulkan.dylib` and fall back to `libMoltenVK.dylib` if that isn't found.  What this means is that the Vulkan SDK will be used if it's installed, which allows validation layers to work.  But if LÖVR is run on a macOS system without the Vulkan SDK, the included MoltenVK library will be used instead.

So far I've only been able to test this by looking at the folder structure output by the CI, so I'd be interested in hearing if anyone has success/failure with these new builds.

I looked into universal macOS builds a bit but couldn't  immediately get them working.  I opened #633 to track this as it's a separate issue.
